### DISCUSSION TITLE
[Question][Hw2]

### DISCUSSION THREAD

**Discussion by LMMilliken** [prosyslab-classroom/cs424-program-reasoning#273] [Q&A]
Name: Louis

I have started the pl part of the homework, and am comparing the output of my implementation with the expected results for the test cases. There are some cases where my implementation outputs satisfiable as expected, but it then also sometimes prints a model that satisfies the formula in cases where it is not expected.
From the readme, `Z3 may not print if the formula is always satisfiable by any model.`, does this mean that it is incorrect to print a satisfying model in cases where all models satisfy the formula?

**Comment by goodtaeeun** [comment]
Hi, can you elaborate on the following sentence?
`There are some cases where my implementation outputs satisfiable as expected, but it then also sometimes prints a model that satisfies the formula in cases where it is not expected.`

**Comment by LMMilliken** [comment]
Sure, for example the output i get when i run `./pl test/pl1.scm` is:
```
satisfiable
(define-fun Q () Bool
  true)
```
Whereas the contents in `pl1.expected` is:
```satisfiable```
So in this case my output does not match the test case due to it printing the model in a case where it didn't need to.

**Comment by goodtaeeun** [comment]
Ah, we have only put `satisfiable` and `unsatisfiable` in the `*.expected` files. This is because the model depends on the specific implementation. If you check the `test/dune` file, you will see that the testing script only checks the first line of your output with `head -n 1` command.

In short, as long as you produce the right result among `satisfiable` and `unsatisfiable`, you are good.

**Comment by LMMilliken** [comment]
Thank you!
